service.go

```diff
@@ -79,20 +80,63 @@ func (s *service) RenderManifests(
 		request: req,
 	}
 
-	if rc.repo, err = git.Clone(
-		rc.request.RepoURL,
-		git.RepoCredentials{
-			SSHPrivateKey: rc.request.RepoCreds.SSHPrivateKey,
-			Username:      rc.request.RepoCreds.Username,
-			Password:      rc.request.RepoCreds.Password,
-		},
-	); err != nil {
-		return res, fmt.Errorf("error cloning remote repository: %w", err)
+	if rc.request.LocalInPath != "" {
+
+		// We'll be taking our input from a local directory which is presumably
+		// a git repository with the desired source commit already checked out.
+		//
+		// This is mainly useful when Kargo proper wishes to handle the reading and
+		// writing to/from remote repositories itself, leaving Kargo Render to
+		// handle rendering only.
+
+		if rc.repo, err = git.CopyRepo(
+			rc.request.LocalInPath,
+			git.RepoCredentials(rc.request.RepoCreds),
+		); err != nil {
+			return res, fmt.Errorf("error copying local repository: %w", err)
+		}
+		// Check if the working tree is dirty
+		var isDirty bool
+		if isDirty, err = rc.repo.HasDiffs(); err != nil {
+			return res, fmt.Errorf("error checking for diffs: %w", err)
+		}
+		if isDirty {
+			return res, errors.New("working tree is dirty; refusing to proceed")
+		}
+		// Check that there is exactly one remote and it's named "origin"
+		var remotes []string
+		if remotes, err = rc.repo.Remotes(); err != nil {
+			return res, fmt.Errorf("error getting remotes: %w", err)
+		}
+		if len(remotes) != 1 || remotes[0] != "origin" {
```

Comment:
Having seen `"origin"` references in multiple places now. Potential indication this should maybe be a public `const` in `pkg/git`.

Reply:
You're right. It snuck up on me because they didn't all happen at once. I'll fix this.

Reply:
Fixed in https://github.com/akuity/kargo-render/pull/261/commits/faa250de19646ae026e6ce99bd3f76fececa7e30